packages/grid_client/scripts/orchestrators/kubernetes_with_qsfs.ts

```diff
@@ -42,45 +42,55 @@ async function main() {
   const grid3 = await getClient(`kubernetes/${name}`);
 
   const qsfs_name = "testQsfsK8sq1";
+  const disk_size = 1;
+  const count = 8;
 
-  const masterQueryOptions: FilterOptions = {
+  const options: FilterOptions = {
     cru: 2,
     mru: 2, // GB
     sru: 6,
     availableFor: grid3.twinId,
     farmId: 1,
   };
 
-  const workerQueryOptions: FilterOptions = {
-    cru: 1,
-    mru: 1, // GB
-    sru: 3,
-    availableFor: grid3.twinId,
-    farmId: 1,
-  };
-
   const qsfsQueryOptions: FilterOptions = {
-    hru: 15,
+    hru: count * disk_size,
     availableFor: grid3.twinId,
     farmId: 1,
   };
 
   const qsfsNodes: number[] = [];
+  let usedResources = 0;
+
+  const nodes = await grid3.capacity.filterNodes(qsfsQueryOptions);
+
+  // Loop over filtered nodes and push in qsfsNodes until they have the required resources
+  for (const node of nodes) {
+    const freeHRU = (node.total_resources.hru - node.used_resources.hru) / 1024 ** 3;
+
+    if (freeHRU >= disk_size) {
```

Comment:
can you please explain the logic behind this conditional?

Reply:
If node's free hru is >= the disk size of the qsfs, then push the node to qsfs nodes array and accumulate usedResources.

Reply:
what is count?

Reply:
It's the number of qsfs deployments. So we need x deployments with the same exact resources mentioned in the qsfs options.